pkg/controller/command/vdri/command.go

```diff
@@ -101,6 +136,132 @@ func (o *Command) CreatePublicDID(rw io.Writer, req io.Reader) command.Error {
+	}
+
+	command.WriteNillableResponse(rw, nil, logger)
+
+	logutil.LogDebug(logger, commandName, saveDIDCommandMethod, "success")
+
+	return nil
+}
+
+// GetDID retrieves the did from the store.
+func (o *Command) GetDID(rw io.Writer, req io.Reader) command.Error {
+	var request IDArg
+
+	err := json.NewDecoder(req).Decode(&request)
+	if err != nil {
+		logutil.LogInfo(logger, commandName, getDIDCommandMethod, err.Error())
+		return command.NewValidationError(InvalidRequestErrorCode, fmt.Errorf("request decode : %w", err))
+	}
+
+	if request.ID == "" {
+		logutil.LogDebug(logger, commandName, getDIDCommandMethod, errEmptyDIDID)
+		return command.NewValidationError(InvalidRequestErrorCode, fmt.Errorf(errEmptyDIDID))
+	}
+
+	didDoc, err := o.didStore.GetDID(request.ID)
+	if err != nil {
+		logutil.LogError(logger, commandName, getDIDCommandMethod, "get did doc: "+err.Error(),
+			logutil.CreateKeyValueString(didID, request.ID))
+
+		return command.NewValidationError(GetDIDErrorCode, fmt.Errorf("get did doc: %w", err))
+	}
+
+	docBytes, err := didDoc.JSONBytes()
+	if err != nil {
+		logutil.LogError(logger, commandName, getDIDCommandMethod, "marshal did doc: "+err.Error(),
+			logutil.CreateKeyValueString(didID, request.ID))
+
+		return command.NewValidationError(GetDIDErrorCode, fmt.Errorf("marshal did doc: %w", err))
+	}
+
+	command.WriteNillableResponse(rw, &Document{
+		DidDocument: string(docBytes),
+	}, logger)
+
+	logutil.LogDebug(logger, commandName, getDIDCommandMethod, "success",
+		logutil.CreateKeyValueString(didID, request.ID))
+
+	return nil
+}
+
+// GetDIDs retrieves the did doc containing name and didID.
+func (o *Command) GetDIDs(rw io.Writer, req io.Reader) command.Error {
+	didRecords := o.didStore.GetDIDs()
+
+	command.WriteNillableResponse(rw, &DIDRecordResult{
+		Result: didRecords,
+	}, logger)
+
+	logutil.LogDebug(logger, commandName, getDIDsCommandMethod, "success")
+
+	return nil
+}
+
+// GetDIDByName retrieves the did by name from the store.
+func (o *Command) GetDIDByName(rw io.Writer, req io.Reader) command.Error {
+	var request NameArg
+
+	err := json.NewDecoder(req).Decode(&request)
+	if err != nil {
+		logutil.LogInfo(logger, commandName, getDIDByNameCommandMethod, err.Error())
+		return command.NewValidationError(InvalidRequestErrorCode, fmt.Errorf("request decode : %w", err))
+	}
+
+	if request.Name == "" {
+		logutil.LogDebug(logger, commandName, getDIDByNameCommandMethod, errEmptyDIDName)
+		return command.NewValidationError(InvalidRequestErrorCode, fmt.Errorf(errEmptyDIDName))
+	}
+
+	id, err := o.didStore.GetDIDByName(request.Name)
+	if err != nil {
+		logutil.LogError(logger, commandName, getDIDByNameCommandMethod, "get did by name : "+err.Error(),
+			logutil.CreateKeyValueString(didName, request.Name))
+
+		return command.NewValidationError(GetDIDByNameErrorCode, fmt.Errorf("get did by name : %w", err))
+	}
+
+	command.WriteNillableResponse(rw, &didstore.Record{
+		Name: request.Name,
```

Comment:
you are not returning DID? just ID & name?

Reply:
mentioned above

Reply:
removed this function , not required